### PR TITLE
Feature/fix faulty retestidproviderutil

### PR DESCRIPTION
**Pull request by roesslerj**

### CODE REVIEW

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java [hunk 1]
```diff
@@ -22,13 +23,26 @@
...
+import de.retest.ui.descriptors.IdentifyingAttributes;
+import de.retest.ui.descriptors.RetestIdProvider;
 import de.retest.ui.descriptors.RootElement;
 import ru.yandex.qatools.ashot.AShot;
 import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
 import ru.yandex.qatools.ashot.shooting.ViewportPastingDecorator;
 
 public class RecheckSeleniumAdapter implements RecheckAdapter {
 
+	// TODO Remove again after retest 3.0.1 release...
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ08&open=AWaLhDVhcPetVqywBZ08)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ08

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java [hunk 2]
```diff
@@ -22,13 +23,26 @@
...
+	public final static RetestIdProvider idProvider = new RetestIdProvider() {
+		@Override
+		public String getRetestId( final IdentifyingAttributes identifyingAttributes ) {
+			return UUID.randomUUID().toString();
+		}
+
+		@Override
+		public void reset() {}
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1186))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ09&open=AWaLhDVhcPetVqywBZ09)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ09

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java [hunk 3]
```diff
@@ -22,13 +23,26 @@
...
+import de.retest.ui.descriptors.RetestIdProvider;
 import de.retest.ui.descriptors.RootElement;
 import ru.yandex.qatools.ashot.AShot;
 import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
 import ru.yandex.qatools.ashot.shooting.ViewportPastingDecorator;
 
 public class RecheckSeleniumAdapter implements RecheckAdapter {
 
+	// TODO Remove again after retest 3.0.1 release...
+	public final static RetestIdProvider idProvider = new RetestIdProvider() {
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Reorder the modifiers to comply with the Java Language Specification. ([squid:ModifiersOrderCheck](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AModifiersOrderCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ07&open=AWaLhDVhcPetVqywBZ07)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ07

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java [hunk 4]
```diff
@@ -22,13 +23,28 @@
...
+import de.retest.ui.descriptors.IdentifyingAttributes;
+import de.retest.ui.descriptors.RetestIdProvider;
 import de.retest.ui.descriptors.RootElement;
 import ru.yandex.qatools.ashot.AShot;
 import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
 import ru.yandex.qatools.ashot.shooting.ViewportPastingDecorator;
 
 public class RecheckSeleniumAdapter implements RecheckAdapter {
 
+	// TODO Remove again after retest 3.0.1 release...
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ08&open=AWaLhDVhcPetVqywBZ08)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ08

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java [hunk 5]
```diff
@@ -22,13 +23,28 @@
...
+import de.retest.ui.descriptors.IdentifyingAttributes;
+import de.retest.ui.descriptors.RetestIdProvider;
 import de.retest.ui.descriptors.RootElement;
 import ru.yandex.qatools.ashot.AShot;
 import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
 import ru.yandex.qatools.ashot.shooting.ViewportPastingDecorator;
 
 public class RecheckSeleniumAdapter implements RecheckAdapter {
 
+	// TODO Remove again after retest 3.1.0 release...
```
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ08&open=AWaLhDVhcPetVqywBZ08)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ08

**Comment by roesslerj** [comment]
I tested manually. We get a new and unique retestId every time. And since this is a workaround, we don't want to cement it with tests. While there is little to test anyway...

File: src/main/java/de/retest/web/RecheckSeleniumAdapter.java (hunk 5 above)
Comment (retest-sonar-cloud) [inline src/main/java/de/retest/web/RecheckSeleniumAdapter.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/retest/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.retest%3Arecheck-web&pullRequest=78&issues=AWaLhDVhcPetVqywBZ08&open=AWaLhDVhcPetVqywBZ08)

[ISSUE_KEY]: AWaLhDVhcPetVqywBZ08

**Pull request merged**